Use PreferenceKeys file to manage SharedPreferences keys.

Skrilltrax opened the pull request:
## :loudspeaker: Type of change
<!--- Put an `x` in the boxes that apply -->
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## :scroll: Description
Let's just stop using raw strings :)

## :bulb: Motivation and Context
Less chances of breaking code due to mistakes in typing strings again

## :green_heart: How did you test it?
Ran the app and changed a few settings. Still needs one more round of testing though.

## :pencil: Checklist
<!--- Put an `x` in the boxes that apply -->
- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

## :crystal_ball: Next steps
Port viscerion's application preferences.

## :camera_flash: Screenshots / GIFs
<!--- Mandatory for UI changes -->

msfjarvis commented on app/src/main/java/com/zeapo/pwdstore/utils/PreferenceKeys.kt:
```diff
@@ -0,0 +1,43 @@
+    const val PREF_KEY_PWGEN_TYPE = "pref_key_pwgen_type"
+    const val OPENPGP_KEY_IDS_SET = "openpgp_key_ids_set"
+
+}
```
needs a trailing newline, run a format pass with Android Studio.

Skrilltrax replied:
Cool, need to update a lot more files though.

Skrilltrax commented:
I think I've updated all the references of the keys. I've updated the PreferenceKeys with `region` and `endregion`, do let me know if you don't like it in the codebase.

msfjarvis commented:
> I think I've updated all the references of the keys. I've updated the PreferenceKeys with `region` and `endregion`, do let me know if you don't like it in the codebase.

I'm not a huge fan of the comments so I'd prefer reverting it. Also bring your branch up-to-date with `develop` since that's enforced by branch protection.

msfjarvis commented:
Completely missed some files, and PasswordStore.kt has a `PREFERENCE_SEEN_AUTOFILL_ONBOARDING` field that needs to go into `PreferenceKeys` as well.

```
app/src/main/java/com/zeapo/pwdstore/Application.kt:        AppCompatDelegate.setDefaultNightMode(when (prefs?.getString("app_theme", getString(R.string.app_theme_def))) {
app/src/main/java/com/zeapo/pwdstore/crypto/DecryptActivity.kt:                            val showPassword = settings.getBoolean("show_password", true)
app/src/main/java/com/zeapo/pwdstore/crypto/DecryptActivity.kt:                            val showExtraContent = settings.getBoolean("show_extra_content", true)
app/src/main/java/com/zeapo/pwdstore/crypto/DecryptActivity.kt:                            if (settings.getBoolean("copy_on_decrypt", true)) {
app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.kt:        val defValue = if (settings!!.getBoolean(PreferenceKeys.AUTOFILL_DEFAULT, true)) "/first" else "/never"
app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.kt:        // if autofill_default is checked and prefs.getString DNE, 'Automatically match with password'/"first" otherwise "never"
app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.kt:        val defValue = if (settings!!.getBoolean("autofill_default", true)) "/first" else "/never"
app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.kt:                        val ttl = Integer.parseInt(settings!!.getString(PreferenceKeys.GENERAL_SHOW_TIME, "45")!!)
```

Skrilltrax commented:
Oh thanks, I'll revisit all the files again. IntelliJ's search results aren't that good when you have 100+ matching items.
